src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -217,10 +217,18 @@ public bool ReportErrors(IEnumerable<Diagnostic> diagnostics, TextWriter console
                     // Not reported from the command-line compiler.
                     continue;
                 }
-
-                consoleOutput.WriteLine(DiagnosticFormatter.Format(diag, this.Culture));
+                
                 ErrorLogger.LogDiagnostic(diag, this.Culture, errorLogger);
 
+                // We want to report diagnostics with source suppression in the error log file.
+                // However, these diagnostics should not be reported on the console output.
+                if (diag.HasSourceSuppression)
```

Comment:
We log "isSuppressedInSource = true" for such diagnostics in the SARIF format errorlog file.